### DISCUSSION TITLE
Routines to count FLOPs whenever it makes sense

### DISCUSSION THREAD

**Discussion by weslleyspereira** [tlapack/tlapack#321] [Ideas]
Currently, we have a few routines that count flops in https://github.com/tlapack/tlapack/blob/master/include/tlapack/starpu/tasks.hpp. They are used by the runtime StarPU to help in the task scheduling. Moreover, some examples measuring performance also use their own FLOP computation. My message here is: the way of measuring FLOPs in the library is incomplete and decentralized.

What I propose (I didn't think too much, so for sure this can be improved):

1. Have three functions to measure FLOPs of each routine when it makes sense to have it:
    addition FLOPs, multiplication FLOPs, and FMA FLOPs.
    The latter may be too much, and too complicated to obtain.
2. Make all parts of the library to use the same FLOP counters.
3. I would start by putting together the FLOPs of the BLAS routines.

**Comment by langou** [comment]
See as well the famous [Appendix C of LAPACK Working Note 41](https://www.netlib.org/lapack/lawnspdf/lawn41.pdf)!
